hibernate-core/src/main/java/org/hibernate/query/sqm/tree/expression/AsWrapperSqmExpression.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Hibernate, Relational Persistence for Idiomatic Java
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later.
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.query.sqm.tree.expression;
+
+import org.hibernate.query.sqm.SemanticQueryWalker;
+import org.hibernate.query.sqm.SqmExpressible;
+import org.hibernate.query.sqm.tree.SqmCopyContext;
+
+public class AsWrapperSqmExpression<T> extends AbstractSqmExpression<T> {
+	private final AbstractSqmExpression<T> expression;
+
+	public AsWrapperSqmExpression(AbstractSqmExpression<T> expression, Class<T> type) {
+		super(
+				expression.nodeBuilder().getTypeConfiguration().getBasicTypeForJavaType( type ),
+				expression.nodeBuilder()
+		);
+		this.expression = expression;
+	}
+
+	AsWrapperSqmExpression(SqmExpressible<T> type, AbstractSqmExpression<T> expression) {
+		super( type, expression.nodeBuilder() );
+		this.expression = expression;
+	}
+
+	@Override
+	public <X> X accept(SemanticQueryWalker<X> walker) {
+		return walker.visitAsWrapperExpression( this );
+	}
+
+	@Override
+	public void appendHqlString(StringBuilder sb) {
+		expression.appendHqlString( sb );
```

Comment:
Until we come up with a proper name for this in HQL I think this should render to a `cast`.

```suggestion
		sb.append( "cast(" );
		expression.appendHqlString( sb );
		sb.append( " as " );
		sb.append( type.getJavaType().getName() );
		sb.append( ')' );
```

Reply:
why? may be I'm missing something but I don't understand this

Reply:
Because this operation is like a cast that can be optimized away.